include/opt.zig

```diff
@@ -0,0 +1,292 @@
+const std = @import("std");
+const assert = std.debug.assert;
+const warn = std.debug.warn;
+const ArrayList = std.ArrayList;
+const Allocator = std.mem.Allocator;
+
+// TODO make tests.  Idk how casting from consts works so no tests for now.
+
+//test "args with no defined flags" {
+//    const args = [_][]const u8{
+//        "test", "arg1",  "--flag1", "arg2",
+//        "--",   "-arg3",
+//    };
+//    var flags = [_]Flag{};
+//    var it = FlagIterator.init(flags[0..0], args[0..]);
+//    assert(std.mem.eql(u8, it.next_arg(), "arg1"));
+//    assert(false);
+//}
+
+const ArgTypeTag = enum {
+    String,
+    None,
+};
+
+const ArgType = union(ArgTypeTag) {
+    String: ?[]u8,
+    None: void,
+};
+
+const Flag = struct {
+    // Unique "ID" for flag (use enum)
+    name: u16,
```

Comment:
This is kind of inconvenient, since users have to tag their flag enum as u16 every time.  Maybe usize would make more sense/be easier to remember?

Reply:
Actually there's probably a way to do this generically. Probably worth looking into eventually